src/FsAutoComplete.Core/SymbolCache.fs

```diff
@@ -9,6 +9,88 @@ open System.Net
 open System.IO
 open Newtonsoft.Json
 
+[<CLIMutable>]
+type SymbolUseRange = {
+    FileName: string
+    StartLine: int
+    StartColumn: int
+    EndLine: int
+    EndColumn: int
+    IsFromDefinition: bool
+    IsFromAttribute : bool
+    IsFromComputationExpression : bool
+    IsFromDispatchSlotImplementation : bool
+    IsFromPattern : bool
+    IsFromType : bool
+    SymbolFullName: string
+    SymbolDisplayName: string
+    SymbolIsLocal: bool
+}
+
+
+module PersistenCacheImpl =
+    open Microsoft.Data.Sqlite
+    open Dapper
+    open System.Data
+
+    let mutable connection : SqliteConnection option = None
+
+    let insert (connection: SqliteConnection) file (sugs: SymbolUseRange[]) =
+        if connection.State <> ConnectionState.Open then connection.Open()
+        use tx = connection.BeginTransaction()
+        let delCmd = sprintf "DELETE FROM Symbols WHERE FileName=\"%s\"" file
+        let inserCmd =
+            sprintf "INSERT INTO SYMBOLS(FileName, StartLine, StartColumn, EndLine, EndColumn, IsFromDefinition, IsFromAttribute, IsFromComputationExpression, IsFromDispatchSlotImplementation, IsFromPattern, IsFromType, SymbolFullName, SymbolDisplayName, SymbolIsLocal) VALUES
+            (@FileName, @StartLine, @StartColumn, @EndLine, @EndColumn, @IsFromDefinition, @IsFromAttribute, @IsFromComputationExpression, @IsFromDispatchSlotImplementation, @IsFromPattern, @IsFromType, @SymbolFullName, @SymbolDisplayName, @SymbolIsLocal)"
+        connection.Execute(delCmd, transaction = tx) |> ignore
+        connection.Execute(inserCmd, sugs, transaction = tx) |> ignore
+        tx.Commit()
+
+    let loadAll (connection: SqliteConnection) =
+        if connection.State <> ConnectionState.Open then connection.Open()
+        let q = "SELECT * FROM SYMBOLS"
+        let res = connection.Query<SymbolUseRange>(q)
+        res
+
+    let loadFile (connection: SqliteConnection) file =
+        if connection.State <> ConnectionState.Open then connection.Open()
+        let q = sprintf "SELECT * FROM SYMBOLS WHERE FileName=\"%s\"" file
+        let res = connection.Query<SymbolUseRange>(q)
+        res
+
+    let initializeCache dir =
+        let connectionString = sprintf "Data Source=%s/.ionide/symbolCache.db" dir
+
+        let dir = Path.Combine(dir, ".ionide")
+        do if not (Directory.Exists dir) then Directory.CreateDirectory dir |> ignore
+        let dbPath = Path.Combine(dir, "symbolCache.db")
+        let dbExists = File.Exists dbPath
+        let connection = new SqliteConnection(connectionString)
+
+        do if not dbExists then
+            let fs = File.Create(dbPath)
+            fs.Close()
+            let cmd = "CREATE TABLE Symbols(
```

Comment:
this doesnt work if the schema will be updated later to v2.
for example an older symbolcache exists on file, and that's will be used but schema is v1